contracts/registries/LicenseRegistry.sol

```diff
@@ -297,6 +301,48 @@ contract LicenseRegistry is ILicenseRegistry, AccessManagedUpgradeable, UUPSUpgr
         return _getLicensingConfig(licensorIpId, licenseTemplate, licenseTermsId);
     }
 
+    /// @notice Verifies the group can add given IP.
+    /// @param groupId The address of the group.
+    /// @param groupRewardPool The address of the reward pool of the group.
+    /// @param ipId The address of the IP to be added to the group.
+    /// @param groupLicenseTemplate the address of the license template attached to the group.
+    /// the IP must have this license template.
+    /// @param groupLicenseTermsId The ID of the license terms attached to the group.
+    /// the IP must have this license terms.
+    /// @return ipLicensingConfig The configuration for license attached to the IP.
+    function verifyGroupAddIp(
+        address groupId,
+        address groupRewardPool,
+        address ipId,
+        address groupLicenseTemplate,
+        uint256 groupLicenseTermsId
+    ) external view returns (Licensing.LicensingConfig memory ipLicensingConfig) {
+        // check if the IP has the same license terms as the group
+        if (!_hasIpAttachedLicenseTerms(ipId, groupLicenseTemplate, groupLicenseTermsId)) {
+            revert Errors.GroupingModule__IpHasNoGroupLicenseTerms(ipId, groupLicenseTemplate, groupLicenseTermsId);
+        }
+        Licensing.LicensingConfig memory lct = _getLicensingConfig(ipId, groupLicenseTemplate, groupLicenseTermsId);
+        if (lct.disabled) {
+            revert Errors.GroupingModule__IpLicenseDisabled(ipId, groupLicenseTemplate, groupLicenseTermsId);
+        }
+        if (lct.expectGroupRewardPool == address(0)) {
+            revert Errors.GroupingModule__IpExpectGroupRewardPoolNotSet(ipId);
+        }
+        if (lct.expectGroupRewardPool != address(groupRewardPool)) {
+            revert Errors.GroupingModule__IpExpectGroupRewardPoolNotMatch(
+                ipId,
+                lct.expectGroupRewardPool,
+                groupId,
+                address(groupRewardPool)
+            );
+        }
+        // IP must not have expiration time to be added to group
+        if (_getExpireTime(ipId) != 0) {
+            revert Errors.GroupingModule__CannotAddIpWithExpirationToGroup(ipId);
```

Comment:
Since this function belongs to `LicenseRegistry`, should we update the error names to start with `LicenseRegistry` instead of `GroupingModule`?

For example, `GroupingModule__IpHasNoGroupLicenseTerms` could be renamed to `LicenseRegistry__IpHasNoGroupLicenseTerms`.